### ISSUE TITLE
Sentence data loader speed

### ISSUE THREAD

**Issue by bmwas** [UKPLab/sentence-transformers#278]
Hello, Thank you for this great library. I was wondering if there is a way to make the sentence data loader a little faster. My triplet loss training data is about 4Gb and taking about 5 hours to load. I have set number of threads to 32 (i.e. torch.set_num_threads(32) ) and my machine has 32 cores/128Gb memory. This has not made the data loading process any faster either. 

 Is there possibly another way of loading the data a little faster or am I missing something? 

All in all I have managed to train and triplet model and it works great but looking for ways of speeding up the data loading process. Any insights are highly appreciated. Thank you!

**Comment by nreimers** [comment]
Hi @bmwas 
The tokenizer is sadly slow and does not run in parallel.

Have a look at this great PR (which I sadly hadn't time yet to integrate):
https://github.com/UKPLab/sentence-transformers/pull/214

It replaces the slow python tokenizer with the faster Rust tokenizer. Pre-processing should be much faster than.

Please report back if the PR works and what the improvement is.

I hope to integrate this Rust Tokenizer soon in this framework.

Best
Nils Reimers

**Comment by bmwas** [comment]
@nreimers Thank you so much. I will look into the new PR.
